src/ts/helper/AdHelper.ts

```diff
@@ -78,29 +107,75 @@ export class AdHelper {
     const adInfo: Conviva.ConvivaMetadata = {
       'c3.ad.id': ad.id,
       'c3.ad.technology': Conviva.Constants.AdType.CLIENT_SIDE,
-      'c3.ad.position': adPosition,
+      'c3.ad.position': AdHelper.mapClientSideAdPosition(adBreakEvent.adBreak, player),
       'c3.ad.system': adSystemName,
       'c3.ad.creativeId': creativeId,
       'c3.ad.firstAdId': firstAdId,
-      'c3.ad.mediaFileApiFramework': mediaFileApiFramework,
-      'c3.ad.firstAdSystem': firstAdSystem,
-      'c3.ad.firstCreativeId': firstCreativeId,
+      [Conviva.Constants.ASSET_NAME]: adTitle,
+      [Conviva.Constants.STREAM_URL]: ad.mediaFileUrl || 'NA',
+
+      // TODO This is not exposed currently. Add it whenever the player
+      // exposes it. Related discussion https://bitmovin.slack.com/archives/C0LJ16JBS/p1716801970037469.
+      'c3.ad.mediaFileApiFramework': 'NA',
+
+      // TODO these two are not exposed currently. Add them whenever the player
+      // exposes them similar to https://github.com/bitmovin-engineering/player-android/pull/3147.
+      // Related discussion https://bitmovin.slack.com/archives/C0LJ16JBS/p1716801796326889.
+      'c3.ad.firstAdSystem': 'NA',
+      'c3.ad.firstCreativeId': 'NA',
+
 
-      // These two are not relevant for the client side (keep in the code for documentation purposes)
+      // These are not relevant for the client side (keep in the code for documentation purposes)
       // 'c3.ad.adStitcher': undefined,
       // 'c3.ad.isSlate': undefined,
     };
 
-    if (adTitle) {
-      adInfo[Conviva.Constants.ASSET_NAME] = adTitle;
+    if ('duration' in ad && ad.duration) {
+      adInfo[Conviva.Constants.DURATION] = ad.duration;
     }
 
-    if (ad.mediaFileUrl) {
-      adInfo[Conviva.Constants.STREAM_URL] = ad.mediaFileUrl;
-    }
+    return adInfo;
+  }
 
-    if ('duration' in ad && ad.duration) {
-      adInfo[Conviva.Constants.DURATION] = ad.duration;
+  public static convertServerSideAdInfoToConvivaAdInfo(serverSideAdInfo: ServerSideAdInfo, allCurrentContentMetadata: Conviva.ConvivaMetadata): Conviva.ConvivaMetadata {
+    const keysToPick = [
+      INTEGRATION_VERSION_CONTENT_METADATA_CUSTOM_TAG,
```

Comment:
On mobile we also copy the custom `streamType` tag.

Reply:
There is no such tag on web explicitly:

<img width="457" alt="image" src="https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/assets/85180621/2afb95a7-6407-48e0-914a-58dee30a5f0a">

But `IS_LIVE` is resolved to `streamType`:

<img width="315" alt="image" src="https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/assets/85180621/39b6022c-c2c4-4852-aad1-2fc10abbc91f">

I checked it back then and verified that it worked as expected. `IS_LIVE` is resolved to `streamType` and from it Conviva infers the actual is life flag. 😄

Reply:
The name is a bit ambiguous, it's also used as metadata [here](https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/blob/e607cc8fab8cfea4bb2b88cf39945c3859c33326/src/ts/ConvivaAnalytics.ts#L544) where it contains dash, hls or progressive if I am not mistaken.

Reply:
Ah, I see https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/112/commits/65874d18c824c9d93cb963820bf68771f84ce978.